packages/bp/src/admin/ui/src/workspace/bots/CreateBotModal.tsx

```diff
@@ -16,6 +16,7 @@ export const sanitizeBotId = (text: string) =>
     .toLowerCase()
     .replace(/\s/g, '-')
     .replace(/[^a-z0-9_-]/g, '')
+    .substring(0, 50) // Only keep the first 50 characters
```

Comment:
**NON-BLOCKING-COMMENT**: I suggest we put the `substring` operation before the `replace`s so that these functions don't try to match patterns on characters we're going to throw away anyway.

Furthermore, I don't have the answer at hand so this is why I'll ask; what's the behaviour of `maxLength=50` in terms of input display? Can you actually type in more than 50 characters? I suppose not? If not, then the `substring` operation might be irrelevant, especially since input goes through a Joi schema validator which should return an error or trim the string down to the desired size.

Reply:
> NON-BLOCKING-COMMENT: I suggest we put the substring operation before the replaces so that these functions don't try to match patterns on characters we're going to throw away anyway.

The reason why I put the substring at the end is that this way, it removes fewer chars that would be kept because trailing spaces and special characters are removed with the replace.

Let's say I have this string "123456789 _- 123456789" and replace the substring of 50 chars with 20. Putting the substring before the replace would give us this string "123456789_-1234567" while if it's at the end, we would have this one "123456789_-123456789"

> Furthermore, I don't have the answer at hand so this is why I'll ask; what's the behaviour of maxLength=50 in terms of input display? Can you actually type in more than 50 characters? I suppose not? If not, then the substring operation might be irrelevant, especially since input goes through a Joi schema validator which should return an error or trim the string down to the desired size.

Exactly, the `maxLength=50` makes it impossible for the user to type in more than 50 chars in the input field. But, it's still possible to provide it with a string property of more than 50 chars. That's the reason behind the substring.